From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vincent Strubel <clipos@ssi.gouv.fr>
Date: Mon, 5 Mar 2012 18:26:12 +0100
Subject: [PATCH 31/32] clip: net: Make 127.0.0.1/16, rather than /8

Make 127.0.0.1/16, rather than /8, the default loopback network, thus
freeing some local addresses for other contexts.

Patch: 2002_loopback_classB.patch
---
 include/linux/in.h      | 2 +-
 include/uapi/linux/in.h | 2 +-
 net/ipv4/devinet.c      | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/in.h b/include/linux/in.h
index 31b493734763d..8ce8dde48ff34 100644
--- a/include/linux/in.h
+++ b/include/linux/in.h
@@ -41,7 +41,7 @@ static inline int proto_ports_offset(int proto)
 
 static inline bool ipv4_is_loopback(__be32 addr)
 {
-	return (addr & htonl(0xff000000)) == htonl(0x7f000000);
+	return (addr & htonl(0xffff0000)) == htonl(0x7f000000);
 }
 
 static inline bool ipv4_is_multicast(__be32 addr)
diff --git a/include/uapi/linux/in.h b/include/uapi/linux/in.h
index eaf94919291aa..50f82af1cc867 100644
--- a/include/uapi/linux/in.h
+++ b/include/uapi/linux/in.h
@@ -283,7 +283,7 @@ struct sockaddr_in {
 
 /* Address to loopback in software to local host.  */
 #define	INADDR_LOOPBACK		0x7f000001	/* 127.0.0.1   */
-#define	IN_LOOPBACK(a)		((((long int) (a)) & 0xff000000) == 0x7f000000)
+#define	IN_LOOPBACK(a)		((((long int) (a)) & 0xffff0000) == 0x7f000000)
 
 /* Defines for Multicast INADDR */
 #define INADDR_UNSPEC_GROUP   	0xe0000000U	/* 224.0.0.0   */
diff --git a/net/ipv4/devinet.c b/net/ipv4/devinet.c
index 7622e2c54bedc..ac55402d2a5fd 100644
--- a/net/ipv4/devinet.c
+++ b/net/ipv4/devinet.c
@@ -1452,8 +1452,8 @@ static int inetdev_event(struct notifier_block *this, unsigned long event,
 				INIT_HLIST_NODE(&ifa->hash);
 				ifa->ifa_local =
 				  ifa->ifa_address = htonl(INADDR_LOOPBACK);
-				ifa->ifa_prefixlen = 8;
-				ifa->ifa_mask = inet_make_mask(8);
+				ifa->ifa_prefixlen = 16;
+				ifa->ifa_mask = inet_make_mask(16);
 				in_dev_hold(in_dev);
 				ifa->ifa_dev = in_dev;
 				ifa->ifa_scope = RT_SCOPE_HOST;
-- 
2.17.0

